### PR TITLE
OGM-359 support stored procedures update2

### PR DESCRIPTION
**Pull request by schernolyas**
[OGM-359](https://hibernate.atlassian.net/browse/OGM-359)

The PR is upgraded version of PR (#916) .

### CODE REVIEW

**Comment by DavideD** [comment]
My bad, I didn't fetch the  repository with the latest changes

**Comment by DavideD** [comment]
Ok, it started to review this but I noticed some things the are not quire right about this implementations.
I'm going to apply some changes.

**Comment by DavideD** [comment]
I'm going to close this one, while reviewing I found several things that are not quite right. I'll send a new PR later

**Comment by DavideD** [comment]
> I'm going to close this one, while reviewing I found several things that are not quite righ

By the way, the problem were in the orignial PR, I just didn't notice them before. This rebase works. Thanks @schernolyas

File: core/src/main/java/org/hibernate/ogm/storedprocedure/impl/NoSQLProcedureResultSetOutputImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class NoSQLProcedureResultSetOutputImpl implements ResultSetOutput {
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+	private final List<?> resultList;
+	private final boolean isResultRefCursor;
+
+	public NoSQLProcedureResultSetOutputImpl(List<?> resultList, boolean isResultRefCursor) {
+		this.resultList = resultList;
+		this.isResultRefCursor = isResultRefCursor;
```
[the remark this answers is not in the document]
Comment (schernolyas) [reply core/src/main/java/org/hibernate/ogm/storedprocedure/impl/NoSQLProcedureResultSetOutputImpl.java]: for using getSingleResult().
The variable equal true if variables of procedure contains REF_CURSOR.

**Pull request closed without merging**